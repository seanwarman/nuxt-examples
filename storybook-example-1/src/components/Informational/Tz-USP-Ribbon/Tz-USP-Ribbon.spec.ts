/**
 * @vitest-environment jsdom
 */

import { mount, VueWrapper, shallowMount } from '@vue/test-utils';
import TzUSPRibbon from './Tz-USP-Ribbon.vue';
import { expect, test, describe, beforeEach, it } from 'vitest';

let wrapper: VueWrapper;

type USPRibbon = {
  offerTitles: string[];
};

const titles = ['Free Delivery', '0% APR Finance', 'Price Match'];

describe('USP Ribbon setup', () => {
  beforeEach(() => {
    wrapper = mount(TzUSPRibbon, {
      props: {
        offerTitles: titles
      }
    });
  });
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('renders offer text content', () => {
    wrapper.findAll('div').every((item, index) => {
      expect(item.text()).toContain(titles[index].toUpperCase());
    });
  });
  it('sets a base class', () => {
    expect(wrapper.classes()).toContain('tz-offer-container');
  });
});
