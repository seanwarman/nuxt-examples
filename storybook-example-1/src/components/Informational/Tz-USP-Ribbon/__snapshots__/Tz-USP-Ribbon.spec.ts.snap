// Vitest Snapshot v1

exports[`USP Ribbon setup > renders correctly 1`] = `
<div
  class="tz-offer-container"
>
  
  <div>
    <span
      class="tz-offer-item"
    >
      FREE DELIVERY
    </span>
  </div>
  <div>
    <span
      class="tz-offer-item"
    >
      0% APR FINANCE
    </span>
  </div>
  <div>
    <span
      class="tz-offer-item"
    >
      PRICE MATCH
    </span>
  </div>
  
</div>
`;
